cpp/ql/src/Critical/GlobalUseBeforeInit.ql

```diff
@@ -98,10 +98,25 @@ predicate callReaches(Call call, ControlFlowNode successor) {
   )
 }
 
+// To avoid many false alarms like `static int a = 1;`
+predicate initialisedAtDeclaration(GlobalVariable v) {
+  exists(VariableDeclarationEntry vde |
+    vde = v.getDefinition()
+    and vde.isDefinition()
+  )
+}
+
+// No need to initialize those variables
+predicate isStdlibVariable(GlobalVariable v) {
+  v.getName() = ["stdin", "stdout", "stderr"]
+}
```

Comment:
This ensures that it's not a global variable in a namespace that shadows the actual `stdin`, `stdout`, or `stderr` variables. Doing so would be crazy, but we may as well handle that case 🙂
```suggestion
predicate isStdlibVariable(GlobalVariable v) { v.hasGlobalName(["stdin", "stdout", "stderr"]) }
```

Reply:
Fixed in https://github.com/github/codeql/pull/16149/commits/132bb9f1d6cb630466cd4649404203fc80a857c2